### PR TITLE
[BB-3018] Load balancers should serve the maintenance pages from the local nginx vhosts and not redirect to external domains

### PR DESCRIPTION
**Pull request by shimulch**
This PR fixes the wrong maintenance page issue during scheduled or unscheduled maintenance and during provisioning.

**JIRA tickets**: https://tasks.opencraft.com/browse/BB-3018

~~**Discussions**:~~
 
**Dependencies**: None

~~**Screenshots**: ~~
~~**Sandbox URL**:~~

**Merge deadline**: "None"

**Testing instructions**:
1. Move an instance to maintenance mode, proper maintenance page should show
2. Provisioning page should show from local Nginx instead of using external server IP

~~**Author notes and concerns**:~~

**Reviewers**
- [ ] @lgp171188 

~~**Settings**~~

### CODE REVIEW

**Review by lgp171188 — COMMENTED** [review]
@shimulch, these changes look okay to me but will have to be tested before approval.

File: playbooks/roles/load-balancer-v2/templates/haproxy.cfg.ctmpl
Comment (lgp171188) [inline playbooks/roles/load-balancer-v2/templates/haproxy.cfg.ctmpl:122]: @shimulch, this condition can be combined into the existing if condition with `and` ([docs](https://golang.org/pkg/text/template/#hdr-Functions))

Comment (shimulch) [reply playbooks/roles/load-balancer-v2/templates/haproxy.cfg.ctmpl]: @lgp171188 Thanks. Updated.

**Review by lgp171188 — APPROVED** [review]
:+1: 
- [x] I tested these changes in the stage environment and verified that:
  * When an instance doesn't have active appservers, the provisioning page from the local nginx server is served.
  * When an instance has active appservers, a proper backend configuration is generated and the instance works okay.
  * Both the maintenance backends now show the correct page instead of the error page.
- [ ] I read through the code
- [ ] I checked for accessibility issues
- [ ] Includes documentation

**Pull request merged**
